src/lightning/pytorch/accelerators/ipu.py

```diff
@@ -49,9 +49,9 @@ def parse_devices(devices: int) -> int:
         return devices
 
     @staticmethod
-    def get_parallel_devices(devices: int) -> List[int]:
+    def get_parallel_devices(devices: int) -> List[torch.device]:
```

Comment:
I believe all accelerators should return a list of torch.devices, which is what the strategy expects to get as `strategy.parallel_devices`

Otherwise there can be silent errors where the wrong device type is used.